### PR TITLE
Some small fixes for cmake macos build

### PR DESCRIPTION
**Pull request by NaLiJa**
When building rlottie according to rlottie docs, the libs are in the build subdirectory, so cmake cannot find the inc folder. Adjusted search path.
The sample apps were broken on macos because the assets and data files were not added to the app bundle. Added file copying to bl_samlpes macro. Not sure if this is the preferred cmake way to get the files into the bundle without the need to list each file separately as resource.

At the moment the app bundles are not self contained yet when using libs from the system, e.g. via homebrew like freetype, glfw,... will create a separate PR for this later.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -182,6 +182,8 @@ if(SAMPLES_BACKEND STREQUAL "auto")
 	endif()
 endif()
 
+set(SAMPLES_DIR ${PROJECT_SOURCE_DIR}/Samples CACHE PATH "Path to samples directory.")
```
Comment (mikke89) [inline CMakeLists.txt:185]: This one is already defined below, so I don't see a need to declare it again here.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Primary CMake build target
 /Build/
+/Dependencies/
```
Comment (mikke89) [inline .gitignore:3]: We already have a .gitignore in the dependencies folder, we shouldn't ignore this one here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -662,12 +664,33 @@ endif()
...
+            set(_DESTINATION ${DESTINATION}/${_DESTINATION})
+        endif ()
+        set_property(SOURCE ${RESOURCE} PROPERTY MACOSX_PACKAGE_LOCATION ${_DESTINATION})
+    endforeach (RESOURCE)
+    set(${VAR} ${_LIST} PARENT_SCOPE)
+endfunction()
+
 # Build and link the samples
-macro(bl_sample NAME)
+macro(bl_sample NAME SDIR)
```
Comment (mikke89) [inline CMakeLists.txt:683]: Maybe a more descriptive name than SDIR? I'm not sure what that means, I guess perhaps another way to write samples dir, but that is already taken.

**Review by mikke89 — COMMENTED** [review]
Hey, thanks for another nice pull request!

I don't have any experience with bundles, and I'm not really able to test this. As longs as this mainly affects apple platforms, I'm fine with delegating and trusting users to update and maintain this. So I guess you can consider yourself the maintainer of this part from here-on out ;)

Please be aware of the (massive) CMake changes in #446. Maybe you could sync up with that at some point, and in particular read the [CMake contributing guide](https://github.com/mikke89/RmlUi/blob/72861917ec4b102f48178888622a0c83cf4483db/CONTRIBUTING.md). But I'm perfectly fine with integrating these smaller fixups here first.

By the way, I see that you're including some unrelated/older commits in this PR. It would be helpful if you could rebase with only the new (PR-related) commits on top of master.

**Comment by NaLiJa** [comment]
Hi there,

fixed the above! Yes, the SDIR was referring to the subdir as I need to copy the folder structure to the app bundle. Have renamed it!


I have with little steps started with looking into adding [glfm ](https://github.com/brackeen/glfm) as a backend, but I cannot say yet if this leads to anything useable.

**Comment by mikke89** [comment]
Thanks for the quick update!

> I have with little steps started with looking into adding [glfm ](https://github.com/brackeen/glfm) as a backend, but I cannot say yet if this leads to anything useable.

Sounds interesting, let us know how it goes.

**Pull request merged**